sharedbuffers/mapped_struct.py

```diff
@@ -114,17 +128,44 @@
 
 ctypes_Array = cython.declare(object, ctypes.Array)
 
+if python3:
+    long = int
+    basestring = (bytes, str)
+
+@cython.cfunc
+@cython.inline
+def make_memoryview(a):
+    return memoryview(a).cast("B")
+
 if cython.compiled:
     # Compatibility fix for cython >= 0.23, which no longer supports "buffer" as a built-in type
-    buffer = cython.declare(object, buffer)  # lint:ok
-    from types import BufferType as buffer
+    if python2:
+        buffer = cython.declare(object, buffer)  # lint:ok
+        from types import BufferType as buffer
+    else:
+        buffer = make_memoryview
 
     assert Py_LT == 0
     assert Py_LE == 1
     assert Py_EQ == 2
     assert Py_NE == 3
     assert Py_GT == 4
     assert Py_GE == 5
+else:
+    if python3:
+        buffer = make_memoryview
+
+def buffer_with_offset(data, offset, size=None):
+    if size is not None:
+        if python3:
+            return buffer(data)[offset:offset+size]
+        else:
+            return buffer(data, offset, size)
```

Comment:
Should call make_memoryview directly (in both python versions), to make use of it being cdef. Have the py3 conditionals inside make_memoryview. Been saying it for a while.